### PR TITLE
Simplify observability example and fix typos

### PR DESCRIPTION
**Pull request by emcfarlane**
Highlight required changes needed to support observability and fix typo on `WithInterceptors`. Use the greetv1 example code shown in getting started as the base.

Fixes https://github.com/connectrpc/otelconnect-go/issues/149

Screenshot 2023-11-15 at 11 12 57 AM

### CODE REVIEW

**Review by akshayjshah — APPROVED** [review]
LGTM, but we'll need to fix the issue management workflow :/

**Comment by nicksnyder** [comment]
Workflow is fixed

File: docs/go/observability.md [hunk 1]
```diff
@@ -22,35 +22,27 @@ OpenTelemetry can be quite complex, so this guide assumes that readers are famil
...
-	log.Print(resp)
-}
+// highlight-next-line
+import "connectrpc.com/otelconnect"
+
+path, handler := greetv1connect.NewGreetServiceHandler(
+	greeter,
+	// highlight-start
+	connect.WithInterceptors(
+		otelconnect.NewInterceptor(/* custom options */),
```
Comment (akshayjshah) [inline docs/go/observability.md:32]: Do we need the `/* custom options */` bit? I think I see what you're going for, but I think it suggests that the user _should_ be supplying more options.

Comment (emcfarlane) [reply docs/go/observability.md:32]: Nope, thought it helped readability on how to configure if needed and user can easily remove from copy pasted code. Will remove.

**Review by akshayjshah — APPROVED** [review]

**Pull request merged**